src/threeIdProvider.ts

```diff
@@ -0,0 +1,139 @@
+import { LinkProof } from '3id-blockchain-utils'
+import {
+  HandlerMethods,
+  RequestHandler,
+  RPCError,
+  RPCRequest,
+  RPCResponse,
+  createHandler,
+} from 'rpc-utils'
+
+import { EncryptedMessage } from './crypto'
+import IdentityWallet from './identity-wallet'
+
+type Origin = string | null | undefined
+
+type Context = {
+  provider: ThreeIdProvider
+  origin: Origin
+}
```

Comment:
This is quite a nice way of dealing with origin. Could we do something with the DID provider? We need origin there as well.

Reply:
Sure, I'll make the change